crates/matrix-sdk-ui/src/timeline/inner/mod.rs

```diff
@@ -158,8 +159,11 @@ impl<P: RoomDataProvider> TimelineInner<P> {
     pub(super) async fn subscribe_batched(
         &self,
     ) -> (Vector<Arc<TimelineItem>>, impl Stream<Item = Vec<VectorDiff<Arc<TimelineItem>>>>) {
-        let (items, stream) = self.subscribe().await;
-        let stream = stream.batch_with(self.state.subscribe_lock_release());
+        trace!("Creating timeline items signal");
+        let state = self.state.read().await;
+        // auto-deref to the inner vector's clone method
```

Comment:
(For what it's worth, I would be comfortable removing that comment, expecting that rustc would cause a compile error for the return type of this function, if there was no deref.)